app/code/core/Mage/Directory/Model/Currency.php

```diff
@@ -134,7 +134,7 @@ public function getRate($toCurrency)
     {
         if (is_string($toCurrency)) {
             $code = $toCurrency;
-        } elseif ($toCurrency instanceof Mage_Directory_Model_Currency) {
+        } elseif ($toCurrency instanceof self) {
```

Comment:
will this still work  if you override Mage_Directory_Model_Currency class with custom class?

Reply:
Yes, overries still work ...

```
<?php

class A
{
    public static function test($class)
    {
        if ($class instanceof self) {
            echo 'yes';
        }
    }
}

class B extends A
{}

A::test(new B());
```

Reply:
I think the problem will occur when people will override the class and copy some methods to change them, ending up with code:
```
<?php

class A
{
    public static function test($class)
    {
        if ($class instanceof self) {
            echo 'yes';
        }
    }
}

class B extends A
{
public static function test($class)
    {
        if ($class instanceof self) {
            echo 'yes';
        }
       //some additional code modification
    }
}

B::test(new A());
```

This will yeld wrong results.
So I'm against changing class name to self in "instanceof".
The logic of instanceof is to check whether sth implements a certain interface. Making it dynamic will cause hard to debug issues.

Reply:
Would this break current code? Guess not.

It is just a problem if someone would copy&paste mehtods, w/o thinking about it .... For existing code it should not matter.

Reply:
I agree with this change since it does not break anything and it is up to coders who copy and paste to do so correctly, I think that has always been the case. However, I see it as a minor point. @sreichel what is the case for doing this as it relates to the end goal to use PHPCS and PHPStan? If we don't include this change will it always be throwing up warnings/errors?

Reply:
@colinmollenhour  this is not related to phpCS or phpStan ... it will not throw errors.  Just want to have clean code ...

Reply:
Ok, seems like a minor subjective issue so I will approve as-is.

Reply:
For me, it's just making people life harder without much benefit.
so +1 for the change in general -1 for the "instanceof" change

Reply:
> For me, it's just making people life harder without much benefit.

Mhh ... at the other hand it enforces coders to review their copy&paste-code a bit more. I'd leave this open for the next days and more opinios about that. OK?